### ISSUE TITLE
Error copying when using rest_cache_url

### ISSUE THREAD

**Issue by tylernisonoff** [bazelbuild/bazel#2609]
Please provide the following information. The more we know about your system and use case, the more easily and likely we can help.

### Description of the problem / feature request / question:
the cache url seems to work fine if i have the majority of my project cached, but fails on a clean build. 
```
.........................
INFO: Found 201 targets...
ERROR: /var/asana-config/asana/asana2/third_party/web/BUILD:299:1: error executing shell command: 'cat third_party/web/tslib_bundle.d.ts | grep -v '/// <reference path=' | grep -v '///<reference path=' > bazel-out/host/bin/third_party/web/tslib.d.ts' failed: bash failed: error executing command
  (cd /home/vagrant/.cache/bazel/_bazel_vagrant/a1dee352bc213c3de72700c6468c2f6f/execroot/asana2 && \
  exec env - \
  /bin/bash -c 'cat third_party/web/tslib_bundle.d.ts | grep -v '\''/// <reference path='\'' | grep -v '\''///<reference path='\'' > bazel-out/host/bin/third_party/web/tslib.d.ts'): com.google.devtools.build.lib.shell.AbnormalTerminationException: Process terminated by signal 15.
Unhandled exception thrown during build; message: Unrecoverable error while evaluating node 'ACTION_EXECUTION:action 'Copying //third_party/web:handlebars [for host]' (JsCopy[[] -> [Artifact:[[/home/vagrant/.cache/bazel/_bazel_vagrant/a1dee352bc213c3de72700c6468c2f6f/execroot/asana2]bazel-out/host/bin]third_party/web/handlebars.js]])' (requested by nodes 'ARTIFACT:third_party/web/handlebars.js //third_party/web:handlebars 9060c8699e5a9815803eae61321ce5de (1968342900 854097986)')
INFO: Elapsed time: 24.253s, Critical Path: 1.94s
java.lang.RuntimeException: Unrecoverable error while evaluating node 'ACTION_EXECUTION:action 'Copying //third_party/web:handlebars [for host]' (JsCopy[[] -> [Artifact:[[/home/vagrant/.cache/bazel/_bazel_vagrant/a1dee352bc213c3de72700c6468c2f6f/execroot/asana2]bazel-out/host/bin]third_party/web/handlebars.js]])' (requested by nodes 'ARTIFACT:third_party/web/handlebars.js //third_party/web:handlebars 9060c8699e5a9815803eae61321ce5de (1968342900 854097986)')
	at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:429)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:496)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IndexOutOfBoundsException: index (0) must be less than size (0)
	at com.google.common.base.Preconditions.checkElementIndex(Preconditions.java:310)
	at com.google.common.base.Preconditions.checkElementIndex(Preconditions.java:293)
	at com.google.common.collect.RegularImmutableList.get(RegularImmutableList.java:67)
	at com.google.devtools.build.lib.remote.TreeNodeRepository.buildParentNode(TreeNodeRepository.java:245)
	at com.google.devtools.build.lib.remote.TreeNodeRepository.buildFromActionInputs(TreeNodeRepository.java:235)
	at com.google.devtools.build.lib.remote.RemoteSpawnStrategy.exec(RemoteSpawnStrategy.java:169)
	at com.google.devtools.build.lib.analysis.actions.SpawnAction.internalExecute(SpawnAction.java:264)
	at com.google.devtools.build.lib.analysis.actions.SpawnAction.execute(SpawnAction.java:272)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor.executeActionTask(SkyframeActionExecutor.java:780)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor.prepareScheduleExecuteAndCompleteAction(SkyframeActionExecutor.java:720)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor.access$800(SkyframeActionExecutor.java:101)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.call(SkyframeActionExecutor.java:610)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.call(SkyframeActionExecutor.java:572)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor.executeAction(SkyframeActionExecutor.java:377)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.checkCacheAndExecuteIfNeeded(ActionExecutionFunction.java:411)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.compute(ActionExecutionFunction.java:195)
	at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:370)
```

### Environment info
* Operating System: ubuntu 14.04

* Bazel version (output of `bazel info release`): 

0.4.0 (my project fails to build on higher versions, but tahts another error)

* If `bazel info release` returns "development version" or "(@non-git)", please tell us what source tree you compiled Bazel from; git commit hash is appreciated (`git rev-parse HEAD`):

### Anything else, information or logs or outputs that would be helpful?
(If they are large, please upload as attachment or provide link).

**Comment by tylernisonoff** [comment]
I fixed the issue on my side by explicitly special-casing the case with 0 input files in my skylark rule -- not sure if it should be expected to handle 0 inputs.

**Comment by hermione521** [comment]
@philwo I guess you work on it?

**Comment by ulfjack** [comment]
You're saying it's crashing for actions that have no inputs, right?

**Comment by tylernisonoff** [comment]
Correct

**Comment by ulfjack** [comment]
We shouldn't crash for actions with no inputs. I'll fix this as soon as I get everything else working again.

**Comment by ulfjack** [comment]
FIxed by 7e0cc9e72f2e7dfbcebdbe49af74ed42d0e8f33a, I think.
